manim/mobject/geometry.py

```diff
@@ -1606,6 +1613,9 @@ def construct(self):
             shift_dir -= label.get_right() + DEFAULT_MOBJECT_TO_MOBJECT_BUFFER * RIGHT
         label.shift(shift_dir)
         label.set_color(color)
+        if show_bg_rec:
```

Comment:
This part is redundant no? Passing `add_background_rectangles_to_entries` in `Matrix` already does these.

Reply:
Oh yes, that sounds better actually. Thanks!
I have made the change in the function.